main.py

```diff
@@ -62,7 +78,91 @@ def new_chat_member(update: Update, context: CallbackContext):
 
-print("starting polling")
-updater.start_polling()
-print("online")
-updater.idle()
+@command("warn")
+def warn_member(update: Update, context: CallbackContext):
+	if update.message.from_user.is_bot: # might be redundant
+		return
+	target = get_warn_target(update.message)
+	if target is None:
+		update.message.reply_text(f'Please reply to a message with /warn')
+		return
+	if target.is_bot:
+		update.message.reply_text(f'You cannot warn a bot')
+		return
+	if not is_admin(update.message.chat, update.message.from_user):
+		update.message.reply_text(f'You are not an admin')
+		return
+	userid = target.id
+	warns = db.get_warns(userid) + 1
+	db.set_warns(userid, warns)
+	update.message.reply_to_message.reply_text(
+		f'You recieved a warn!\nNow you have {warns} warns.')
+
+@command("unwarn")
+def unwarn_member(update: Update, context: CallbackContext):
+	if update.message.from_user.is_bot: # might be redundant
+		return
+	target = get_warn_target(update.message)
+	if target is None:
+		update.message.reply_text(f'Please reply to a message with /unwarn')
+		return
+	if target.is_bot:
+		update.message.reply_text(f'Bots do not have warns')
+		return
+	if not is_admin(update.message.chat, update.message.from_user):
+		update.message.reply_text(f'You are not an admin')
+		return
+	userid = target.id
+	warns = db.get_warns(userid)
+	if warns > 0: warns -= 1
+	db.set_warns(userid, warns)
+	reply = f'You\'ve been a good hooman!\n'
+	if warns == 0:
+		reply += 'Now you don\'t have any warns.'
+	else:
+		reply += f'Now you have {warns} warns.'
+	update.message.reply_to_message.reply_text(reply)
+
+@command("clearwarns")
+def unwarn_member(update: Update, context: CallbackContext):
+	if update.message.from_user.is_bot: # might be redundant
+		return
+	target = get_warn_target(update.message)
+	if target is None:
+		update.message.reply_text(f'Please reply to a message with /clearwarns')
+		return
+	if target.is_bot:
+		update.message.reply_text(f'Bots do not have warns')
+		return
+	if not is_admin(update.message.chat, update.message.from_user):
+		update.message.reply_text(f'You are not an admin')
+		return
+	userid = target.id
+	warns = 0
+	db.set_warns(userid, warns)
+	reply = f'You\'ve been a good hooman!\nNow you don\'t have any warns.'
+	update.message.reply_to_message.reply_text(reply)
+
+@command("warns")
+def warns_member(update: Update, context: CallbackContext):
+	if update.message.from_user.is_bot: # might be redundant
+		return
+	target = get_warn_target(update.message)
+	if target is None:
+		userid = update.message.from_user.id
+		warns = db.get_warns(userid)
+		update.message.reply_text(f'You have {warns} warns.')
+		return
+	if target.is_bot:
+		update.message.reply_text(f'Bots do not have warns')
+		return
+	warns = db.get_warns(target.id)
+	update.message.reply_text(f'They have {warns} warns.')
+
+try:
+	print("starting polling")
+	updater.start_polling()
+	print("online")
+	updater.idle()
+finally:
+	db.dump()
```

Comment:
I'd periodically dump the database to be safe, e.g. what if the power goes out?

Reply:
Added a saver thread that saves database every 5 minutes